hs/t/y/child_reach_app.rsh

```diff
@@ -0,0 +1,49 @@
+'reach 0.1';
+
+export const a = Reach.App(() => {
+  // How should scope be handled?
+  // We want to evaluate locally so that
+  // the state says we haven't initialized the app
+  // and we want to be in the correct state.
+  // but do we want to access outer values
+  // like closures can?
+  const B = Participant('A', {});
+  init();
+  B.publish();
+  commit();
+})
+
+const t = Reach.App(() => {
+  // How should scope be handled?
+  // We want to evaluate locally so that
+  // the state says we haven't initialized the app
+  // and we want to be in the correct state.
+  // but do we want to access outer values
+  // like closures can?
+  const B = Participant('A', {});
+  init();
+  B.publish();
+  commit();
+})
+
+export const main = Reach.App(() => {
+  const A = Participant('A', {});
+  init();
+  A.publish();
+  const ctor1 = new Contract(t);
+  const ctor2 = new Contract(Reach.App(() => {
+    // How should scope be handled?
+    // We want to evaluate locally so that
+    // the state says we haven't initialized the app
+    // and we want to be in the correct state.
+    // but do we want to access outer values
+    // like closures can?
```

Comment:
We should be able to access static things (like numbers) and not dynamic things (like A, or variables that are published). So, you could do...
```
A.publish(x);
const y = 5;
const z = x + y;
```
and demonstrate that `ctor2` can look at `y` but not `x` or `z`.

Reply:
I think that you should change this test case so that `a` is `(y) => same-code-as-right-here` and do `a(y)`, then make it so that `t = a(5)`. Right now, we write lots of programs like `t`, but this is a new test to make sure we support things like `a(y)` and embedding.

Reply:
As far as actually implementing it...

Option 1: Scrub the environment of things that aren't allowed
Pro: Simple
Con: You might miss something and you'll let the name be used again

Option 2: Add something to the environment that says what "universe" you are in and then error if you ever look at something from the wrong universe
Pro: Simple conceptually
Con: Big code change:
- Must change the `SLEnv` data type to add another field for the universe of each entry
- Add something to the overall env that has the current universe
- Must include a counter that gets incremented each time you start a new `Reach.App` scope.

I think you should do Option 2